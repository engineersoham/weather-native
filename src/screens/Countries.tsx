/** @format */

import { Button } from "@mui/material";
import axios from "axios";
import { FC, useEffect, useState } from "react";
import { useLocation } from "react-router-dom";

const Countries: FC = () => {
	const [countriesData, setCountriesData] = useState<Record<string, any>[]>();
	const [countryLoading, setCountryLoading] = useState<boolean>(true);
	const [countryError, setCountryError] = useState<boolean>(false);

	const location = useLocation();

	useEffect(() => {
		getCountryData();
	}, []);

	function getCountryData() {
		setCountryLoading(true);
		axios //@ts-ignore
			.get(`https://restcountries.com/v3.1/name/${location?.state?.country}`)
			.then((res) => setCountriesData(res.data))
			.catch(() => setCountryError(true))
			.finally(() => setCountryLoading(false));
	}
	console.log(countriesData?.length);
	return (
		<div
			style={{
				display: "flex",
				alignItems: "center",
				flexDirection: "column",
				height: window.innerHeight,
				padding: 10,
			}}
		>
			<div>
				{countryLoading ? (
					<div>Loading...</div>
				) : countryError ? (
					<div>Something Went wrong!</div>
				) : (
					countriesData?.map((country) => {
						return (
							<div
								style={{
									minWidth: window.innerWidth * 0.4,
									margin: 10,
									padding: 20,
									borderRadius: 20,
									boxShadow: "0px 0px 10px rgba(0,0,0,.5)",
									alignItems: "center",
									flexDirection: "column",
									display: "flex",
								}}
							>
								<div
									style={{
										flexDirection: "row",
										justifyContent: "space-around",
										alignItems: "center",
										display: "flex",
									}}
								>
									<div style={{ margin: 10 }}>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Name {"       "} : <h3> {country?.name?.common}</h3>
										</p>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Capital {"     "}:{" "}
											<h3>
												{"  "} {country?.capital?.[0]}
											</h3>
										</p>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Population {"  "}: <h3>{country?.population}</h3>
										</p>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Latitude {"    "}: <h3>{country?.latlng?.[0]}</h3>
										</p>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Longitude {"   "}: <h3>{country?.latlng?.[1]}</h3>
										</p>
										<p
											style={{
												flexDirection: "row",
												alignItems: "center",
												display: "flex",
												height: 15,
											}}
										>
											Image URL {"   "}: <h3>{country?.flags?.svg}</h3>
										</p>
									</div>
									<img
										src={country?.flags?.svg}
										alt={`${country.name?.common} flag ${country.flags[0]}`}
										width={100}
										height={100}
										style={{ width: 150, height: 100 }}
									/>
								</div>
								<Button
									style={{
										color: "white",
										borderRadius: 10,

										backgroundColor: "blue",
										padding: 10,
									}}
								>
									Capital Weather
								</Button>
							</div>
						);
					})
				)}
			</div>
		</div>
	);
};

export default Countries;
