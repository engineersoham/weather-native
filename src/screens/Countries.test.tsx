/** @format */

import { render, screen } from "@testing-library/react";
import Countries from "./Countries";

jest.mock("react-router-dom", () => {
	return {
		useLocation: () => {
			return {
				state: {
					country: "india",
				},
			};
		},
	};
});

describe("Countries", () => {
	jest.setTimeout(10000);

	it("renders", () => {
		render(<Countries />);
	});

	it("matches snapshot", () => {
		const view = render(<Countries />);
		expect(view).toMatchSnapshot();
	});

	it("has the back button", async () => {
		render(<Countries />);
		const backButton = await screen.findByTestId("backButton");
		backButton.click();
		expect(backButton).toBeInTheDocument();
	});

	it("has progress circle shown", () => {
		render(<Countries />);
		const progress = screen.getByTestId("countriesProgress");
		expect(progress).toBeInTheDocument();
	});

	it("has countries list", async () => {
		render(<Countries />);
		await new Promise((r) => setTimeout(r, 5000));
		const countriesList = screen.getByTestId("countriesList");
		expect(countriesList).toBeInTheDocument();

		const weatherButton = screen.getByTestId("weatherButton");
		weatherButton.click();
		await new Promise((r) => setTimeout(r, 4000));
		const weatherModal = screen.getByTestId("weatherModal");
		expect(weatherModal).toBeInTheDocument();
	});
});
