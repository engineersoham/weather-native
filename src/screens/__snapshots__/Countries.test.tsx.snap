// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        style="padding: 10px; align-items: center; height: 768px; flex-direction: column; display: flex;"
      >
        <div
          style="margin: 20px;"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="backButton"
            style="top: 10px; left: 10px; position: absolute;"
            tabindex="0"
            type="button"
          >
            &lt; Back
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
          data-testid="countriesList"
          style="padding: 10px; align-items: center; height: 768px; flex-direction: column; display: flex;"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
            data-testid="countriesProgress"
            role="progressbar"
            style="width: 40px; height: 40px; position: absolute; top: 50%; bottom: 50%; text-align: center;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      style="padding: 10px; align-items: center; height: 768px; flex-direction: column; display: flex;"
    >
      <div
        style="margin: 20px;"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="backButton"
          style="top: 10px; left: 10px; position: absolute;"
          tabindex="0"
          type="button"
        >
          &lt; Back
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        data-testid="countriesList"
        style="padding: 10px; align-items: center; height: 768px; flex-direction: column; display: flex;"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          data-testid="countriesProgress"
          role="progressbar"
          style="width: 40px; height: 40px; position: absolute; top: 50%; bottom: 50%; text-align: center;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
